### DISCUSSION TITLE
f0b473e1-c892-4a21-8319-68f56f713c0d

### DISCUSSION THREAD

**Discussion by giscus** [SSWConsulting/SSW.Rules.Content#9615] [Rules Discussions]
# f0b473e1-c892-4a21-8319-68f56f713c0d

Generic names like "manager" are easy and can trick us into thinking they are good for consistency. But they can quickly undermine conveying the meaning of your code.

https://www.ssw.com.au/rules/avoid-generic-names/

**Comment by wicksipedia** [comment]
💯 - project names like shared and common come to mind as well, they are horrible since they become a catch all receptacle

**Comment by Calinator444** [comment]
Well I guess it's time for me to rename this then

```
type PageData = {
  filename: string;
  isNewConsultingPage: boolean;
};

**Comment by chrisschultzssw** [comment]
This is important because it's an easy one to miss - generic names will make sense when you create them, but to other devs (or future you) they will not be clear at all.

**Comment by ncn-ssw** [comment]
I regard this as the most important tip, as this can lead to costly misunderstandings about what a section of code achieves.

**Comment by bahjatalaadelssw** [comment]
It's important to take the time to label anything with a meaningful title, it helps with searchability and clarity.

**Comment by thom-wang** [comment]
I used to write things like Helper and Manager a lot because it is generic and you can put everything in it easily. After this I came to realize it is really hard for other devs to read it afterwards and I'll change them to specific meaning names instead.

**Comment by babakamyljanovssw** [comment]
Good naming the services or handlers are really important to understand what functions they are responsible for.

**Comment by danielmackay** [comment]
'Service' is another generic name that we need to be on the lookout for. Although they have their place (e.g. Domain Services), developers should avoid falling into the trap of calling everything a 'Service', because they can't think of a better name.
